src/Symfony/Component/HttpKernel/EventListener/ValidationFailedExceptionListener.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\ExceptionEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Serializer\Exception\NotEncodableValueException;
+use Symfony\Component\Serializer\SerializerInterface;
+use Symfony\Component\Validator\Exception\ValidationFailedException;
+
+/**
+ * @author Konstantin Myakshin <molodchick@gmail.com>
+ */
+final class ValidationFailedExceptionListener implements EventSubscriberInterface
+{
+    public function __construct(private readonly SerializerInterface $serializer)
+    {
+    }
+
+    public function onKernelException(ExceptionEvent $event): void
+    {
+        $throwable = $event->getThrowable();
+        if (!$throwable instanceof ValidationFailedException) {
+            return;
+        }
+
+        $request = $event->getRequest();
+        $format = $request->getPreferredFormat('json');
+
+        try {
+            $data = $this->serializer->serialize($throwable->getViolations(), $format);
+        } catch (NotEncodableValueException) {
+            return;
+        }
+
+        $response = new Response(
+            $data,
+            Response::HTTP_UNPROCESSABLE_ENTITY,
+            ['Content-Type' => $request->getMimeType($format)]
+        );
+        $event->setResponse($response);
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            KernelEvents::EXCEPTION => ['onKernelException', -32],
+        ];
+    }
+}
```

Comment:
Why do you need new exception listeners? Isn't that supposed to be the responsibility of the validation component?

Reply:
`RequestPayloadValueResolver` calls Validator, check for result and throws `ValidationFailedException` for an validation errors. Then we catch this Exception using Listeners mechanism to normalze Violations and respond with proper http code.